src/app/features/activity-list/hooks/use-current-submitted-transactions.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+
+import { useCurrentAccountTxIds } from '@app/query/transactions/transaction.hooks';
+import { submittedTransactionsActions } from '@app/store/submitted-transactions/submitted-transactions.actions';
+import { useSubmittedTransactions } from '@app/store/submitted-transactions/submitted-transactions.selectors';
+import { safelyFormatHexTxid } from '@app/common/utils/safe-handle-txid';
+import { useAppDispatch } from '@app/store';
+
+export function useCurrentSubmittedTransactions() {
+  const submittedTransactions = useSubmittedTransactions();
+  const accountTxIds = useCurrentAccountTxIds();
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    Object.values(submittedTransactions).map(tx => {
+      if (!tx) return;
+      if (accountTxIds.includes(safelyFormatHexTxid(tx.txId)))
+        dispatch(submittedTransactionsActions.transactionRemoved(tx.txId));
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountTxIds, submittedTransactions]);
+
+  return submittedTransactions;
+}
```

Comment:
Really curious to see how we could kill this hook/reimplement another way

Reply:
I'll try to move this today ...seems like it just needs to update when pending txs update.

Reply:
Also, can you check often this logic executes? Looking at `useCurrentAccountTxIds`

https://github.com/hirosystems/stacks-wallet-web/blob/3a890d665a70241a9593859ea889c2c0c897290d/src/app/query/transactions/transaction.hooks.ts#L10-L13

it calls `.map()`, which on every render, will return a new array instance. Then, on every render, `useEffect` here will detect it as a new array, triggering this logic